client/components/ConfigureActiveRuns/index.jsx

```diff
@@ -135,8 +162,118 @@ class ConfigureActiveRuns extends Component {
         });
     }
 
+    showChanges() {
+        this.setState({
+            showChangeModal: true
+        });
+    }
+
+    closeChanges() {
+        this.setState({
+            showChangeModal: false
+        });
+    }
+
+    /** 
+     * This is a naive algorithm for finding all the configuration changes. 
+     * It can be vastly improved, especially with smarter logic on handling
+     * duplicate changes.
+     * */
+    setConfigChanges(newConfig) {
```

Comment:
this should be called "getConfigChanges" or "calculateConfigChanges" because it doesn't set any thing, it just does some data munging and returns the result!